sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/StorageImplUtils.java

```diff
@@ -389,10 +391,27 @@ public Map.Entry<String, String> next() {
                 throw new NoSuchElementException();
             }
 
-            int nextPosition = queryParameters.indexOf('=', position);
+            int nextPosition = position;
```

Comment:
We should consider making the query parsing logic in core public. This is duplicating ImplUtils in core.